src/compiler/utilities.ts

```diff
@@ -1030,6 +1028,21 @@ module ts {
             newEndN = Math.max(newEnd2, newEnd2 + (newEnd1 - oldEnd2));
         }
 
-        return createTextChangeRange(createTextSpanFromBounds(oldStartN, oldEndN), /*newLength: */newEndN - oldStartN);
+        return createTextChangeRange(createSpanFromBounds(oldStartN, oldEndN), /*newLength: */newEndN - oldStartN);
+    }
+
+    export function nodeArrayEnd(array: NodeArray<Node>) {
+        var end = array.length === 0 ? array.start : spanEnd(lastOrUndefined(array));
+        if (!array.hasTrailingComma) {
+            return end;
+        }
+
+        Debug.assert(array.length > 0);

```

Comment:
You can't have an empty array with a trailing comma.  The delimited parsing code won't ever make that.  You only try parsing a comma once you've parsed something real.